airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -406,17 +407,17 @@ def _run_task_kwargs(
 
         One last chance to modify Boto3's "run_task" kwarg params before it gets passed into the Boto3 client.
         """
-        run_task_api = deepcopy(self.run_task_kwargs)
-        container_override = self.get_container(run_task_api["overrides"]["containerOverrides"])
+        run_task_kwargs = deepcopy(self.run_task_kwargs)
+        run_task_kwargs = merge_dicts(run_task_kwargs, exec_config)
```

Comment:
Good call, @Taragolis .   I always forget that helper exists.

@o-nikolas - the [first step of merge_dicts](https://github.com/aws-mwaa/upstream-to-airflow/blob/0a4ed7d557809ad81ecc50d197c33c8d178c42ce/airflow/utils/helpers.py#L208) is to take dict1.copy(), do we still need the deepcopy here on L410?

Reply:
I think it's nice because I don't know who will possibly mettle with that helper function in the future. But I also see your point. Happy to remove it if you feel strongly!

Reply:
Probably doesn't hurt to leave it as a safety net.  Feel free to resolve this.